docs/contributing/contributing-code/contributing-code-control-plane/running-controlplane-locally.md

```diff
@@ -105,17 +106,42 @@ Run this command to create the namespace that will be used to store data.
 kubectl create namespace radius-testing
 ```
 
-## Setup Step 3: Create Resource Group and Environment
+## Setup Step 4: Run Deployment Engine as docker container
+
+> 💡 This step is required only if you are an external contributor and do not have access to `radius-project/deployment-engine` repo.
+> 
+`docker run  -e RADIUSBACKENDURL=http://host.docker.internal:9000/apis/api.ucp.dev/v1alpha3 -p 5017:8080 ghcr.io/radius-project/deployment-engine:latest`
+
+`host.docker.internal` is a special DNS name provided by Docker that allows containers to access services running on the host machine
+
+## Setup Step 5: Create Resource Group and Environment
 
 At this point Radius is working but you don't have a resource group or environment. You can launch Radius and then use the CLI to create these.
 
 In VS Code:
 
 - Open the Debug tab in VS Code
-- Select `Launch Control Plane (all)` from the drop-down
+- If you have access to `radius-project/deployment-engine` repo
+  - Select `Launch Control Plane (all)` from the drop-down 
+- If you are an external contributor, open launch.json and comment out `Launch Deployment Engine` in `Launch Control Plane (all)`. The debug setup will use the Deployment Engine running as docker container. 
+  ```
+  "compounds": [
+    {
+      "name": "Launch Control Plane (all)",
+      "configurations": [
+        "Launch UCP",
+        "Launch Applications RP",
+        "Launch Dynamic RP",
+        "Launch Controller",
+        // "Launch Deployment Engine"
```

Comment:
This is only if we are running Deployment Engine as a container, right? Should we specify that or delete this line all together?